src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/IconTintColor/IconTintColorBehavior.windows.cs

```diff
@@ -140,6 +153,21 @@ void LoadAndApplyImageTintColor(View element, WImage image, Color color)
 
 			ApplyTintColor();
 		}
+		else if (image.IsLoaded)
+		{
+			// Sometimes WImage source doesn't match View source so the image is not ready to be tinted
+			// We must wait for next ImageOpened event
+			if (element is IImageElement { Source: FileImageSource fileImageSource }
+				&& image.Source is BitmapImage bitmapImage
+				&& Uri.Compare(new Uri($"{bitmapImage.UriSource.Scheme}:///{fileImageSource.File}"), bitmapImage.UriSource, UriComponents.Path, UriFormat.Unescaped, StringComparison.OrdinalIgnoreCase) is not 0)
+			{
+				image.ImageOpened += OnImageOpened;
+			}
+			else
+			{
+				ApplyTintColor();
```

Comment:
Could you please update the [`IconTintColorBehaviorPage` in the sample app](https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/Pages/Behaviors/IconTintColorBehaviorPage.xaml) to include a use-case that hits this specific line of code where this specific `else` block resolves?

```cs
else
{
    ApplyTimeColor();
}
```

I cannot reach this branch using the Sample App and am unable to test it and verify it works.